### ISSUE TITLE
getAttachedShaders is failing (WebGL)

### ISSUE THREAD

**Issue by Treata11** [brave/brave-browser#37044]
## Test plan
1. Visit https://www.shadertoy.com/view/X3cXDj
2. Ensure there are graphics showing in the display portion. It should look like this:
    ![image](https://github.com/brave/brave-browser/assets/4733304/c1ab4abd-fe99-480d-8b13-3ca7ae3e2759)

## Description 

If you head to [ShaderToy](shadertoy.com), you can see the shader previews without any issues. But if you select one to view it in brave (like [this example](https://www.shadertoy.com/view/lX2GzD)), they simply won't be shown on your screen. Not even when you go full screen. 

## Steps to Reproduce

   1. Select a desired shader from [ShaderToy](shadertoy.com)
   4. You'll instantly notice that the shader fails to compile

## Actual result:

You'll only see a black screen. The shaders will fail to compile
<img width="1391" alt="Untitled" src="https://github.com/brave/brave-browser/assets/107116406/eca960d0-2371-40a0-993a-37c6942b525a">

## Expected result:
The web used to work fine in the previous version of brave & you were able to see the shader.

## Reproduces how often: 

Easily reproduced.

## Brave version (brave://version info)

**Brave**	1.64.109 Chromium: 123.0.6312.58 (Official Build) (arm64) 
**Revision**	ded360c0dc8a4bd8aecb9cbe32a7aa4477d42246
**OS**	macOS Version 14.4 (Build 23E214)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes.
- Can you reproduce this issue with the beta channel? Yes.
- Can you reproduce this issue with the nightly channel? Haven't tried.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No.
- Does the issue resolve itself when disabling Brave Rewards? Haven't tried.
- Is the issue reproducible on the latest version of Chrome? No.

**Comment by BaRaDa74** [comment]
works fine on chrome,  but doesnt on brave nightly

**Comment by Rex109** [comment]
Bump, i have this problem too and i'm actually about to switch browser once and for all

**Comment by Treata11** [comment]
> Bump, i have this problem too and i'm actually about to switch browser once and for all

It arguably is the best browser I've ever used ... Never bumped to an issue like this before personally while using Brave for almost 3 years now.

**Comment by LeonFedotov** [comment]
please fix this!
maybe theres a flag one needs to flip in the config to make shadertoy work, they are using some internal gl api or something that is restricted by default on brave or what?

**Comment by Diffus3d** [comment]
I've tried disabling fingerprinting, site blockers, and numerous other settings with no luck.  I noticed the console is filled with errors.  Not sure if this is a brave issue or a shadertoy website issue.  

`www.shadertoy.com/:6 
        
        
       GET https://www.googletagmanager.com/gtag/js?id=G-7SYLWGZYJ2 net::ERR_BLOCKED_BY_CLIENT
component.jsx:6 DonateButtonManager props  {env: 'production', hosted_button_id: '8H3QTL2VPKSLL', image: {…}}
(index):1144 WebGL (2.0=true): Asynch Compile: yes, Textures: F32 [yes], F16 [yes], Depth [yes], LOD [yes], Aniso [yes], Units [16], Max Size [16384], Cube Max Size [16384], Targets: MRT [yes], F32 [yes], Max Size [16384]
(index):1144 WebGL (2.0=true): Asynch Compile: yes, Textures: F32 [yes], F16 [yes], Depth [yes], LOD [yes], Aniso [yes], Units [16], Max Size [16384], Cube Max Size [16384], Targets: MRT [yes], F32 [yes], Max Size [16384]
MsjXDm.jpg:1 
        
        
       GET https://www.shadertoy.com/media/shaders/MsjXDm.jpg 404 (Not Found)
Image (async)
previewLoadScreenshot @ (index):8429
iInitUI @ (index):8092
resultsInitStatic @ (index):8120
onload @ (index):8599
(index):1144 WebGL (2.0=true): Asynch Compile: yes, Textures: F32 [yes], F16 [yes], Depth [yes], LOD [yes], Aniso [yes], Units [16], Max Size [16384], Cube Max Size [16384], Targets: MRT [yes], F32 [yes], Max Size [16384]
(index):2060 Uncaught TypeError: Cannot read properties of null (reading '1')
    at me.GetTranslatedShaderSource ((index):2060:81)
    at (index):4508:53
    at checkErrors ((index):2126:37)
    at loopCheckCompletion ((index):2140:41)
me.GetTranslatedShaderSource @ (index):2060
(anonymous) @ (index):4508
checkErrors @ (index):2126
loopCheckCompletion @ (index):2140
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
me.CreateShader @ (index):2144
EffectPass.NewShader @ (index):4493
Effect.NewShader @ (index):7593
(anonymous) @ (index):7750
Effect.Compile @ (index):7748
iLoadAndCompile @ (index):7974
iProcessShader @ (index):8030
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8115
setTimeout (async)
iInitShaders @ (index):8115
resultsInitStatic @ (index):8122
onload @ (index):8599
(index):2060 Uncaught TypeError: Cannot read properties of null (reading '1')
    at me.GetTranslatedShaderSource ((index):2060:81)
    at (index):4508:53
    at checkErrors ((index):2126:37)
    at loopCheckCompletion ((index):2140:41)
me.GetTranslatedShaderSource @ (index):2060
(anonymous) @ (index):4508
checkErrors @ (index):2126
loopCheckCompletion @ (index):2140
setTimeout (async)
me.CreateShader @ (index):2144
EffectPass.NewShader @ (index):4493
Effect.NewShader @ (index):7593
(anonymous) @ (index):7750
Effect.Compile @ (index):7748
iLoadAndCompile @ (index):7974
iProcessShader @ (index):8030
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8115
setTimeout (async)
iInitShaders @ (index):8115
resultsInitStatic @ (index):8122
onload @ (index):8599
(index):1144 WebGL (2.0=true): Asynch Compile: yes, Textures: F32 [yes], F16 [yes], Depth [yes], LOD [yes], Aniso [yes], Units [16], Max Size [16384], Cube Max Size [16384], Targets: MRT [yes], F32 [yes], Max Size [16384]
(index):2060 Uncaught TypeError: Cannot read properties of null (reading '1')
    at me.GetTranslatedShaderSource ((index):2060:81)
    at (index):4508:53
    at checkErrors ((index):2126:37)
    at loopCheckCompletion ((index):2140:41)
me.GetTranslatedShaderSource @ (index):2060
(anonymous) @ (index):4508
checkErrors @ (index):2126
loopCheckCompletion @ (index):2140
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
me.CreateShader @ (index):2144
EffectPass.NewShader @ (index):4493
Effect.NewShader @ (index):7593
(anonymous) @ (index):7750
Effect.Compile @ (index):7748
iLoadAndCompile @ (index):7974
iProcessShader @ (index):8030
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8115
setTimeout (async)
iInitShaders @ (index):8115
resultsInitStatic @ (index):8122
onload @ (index):8599
(index):1144 WebGL (2.0=true): Asynch Compile: yes, Textures: F32 [yes], F16 [yes], Depth [yes], LOD [yes], Aniso [yes], Units [16], Max Size [16384], Cube Max Size [16384], Targets: MRT [yes], F32 [yes], Max Size [16384]
(index):2060 Uncaught TypeError: Cannot read properties of null (reading '1')
    at me.GetTranslatedShaderSource ((index):2060:81)
    at (index):4508:53
    at checkErrors ((index):2126:37)
    at loopCheckCompletion ((index):2140:41)
me.GetTranslatedShaderSource @ (index):2060
(anonymous) @ (index):4508
checkErrors @ (index):2126
loopCheckCompletion @ (index):2140
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
me.CreateShader @ (index):2144
EffectPass.NewShader @ (index):4493
Effect.NewShader @ (index):7593
(anonymous) @ (index):7750
Effect.Compile @ (index):7748
iLoadAndCompile @ (index):7974
iProcessShader @ (index):8030
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8115
setTimeout (async)
iInitShaders @ (index):8115
resultsInitStatic @ (index):8122
onload @ (index):8599
(index):2060 Uncaught TypeError: Cannot read properties of null (reading '1')
    at me.GetTranslatedShaderSource ((index):2060:81)
    at (index):4508:53
    at checkErrors ((index):2126:37)
    at loopCheckCompletion ((index):2140:41)
me.GetTranslatedShaderSource @ (index):2060
(anonymous) @ (index):4508
checkErrors @ (index):2126
loopCheckCompletion @ (index):2140
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
me.CreateShader @ (index):2144
EffectPass.NewShader @ (index):4493
Effect.NewShader @ (index):7593
(anonymous) @ (index):7750
Effect.Compile @ (index):7748
iLoadAndCompile @ (index):7974
iProcessShader @ (index):8030
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8041
setTimeout (async)
iProcessShader @ (index):8041
(anonymous) @ (index):8115
setTimeout (async)
iInitShaders @ (index):8115
resultsInitStatic @ (index):8122
onload @ (index):8599
(index):2060 Uncaught TypeError: Cannot read properties of null (reading '1')
    at me.GetTranslatedShaderSource ((index):2060:81)
    at (index):4508:53
    at checkErrors ((index):2126:37)
    at loopCheckCompletion ((index):2140:41)
me.GetTranslatedShaderSource @ (index):2060
(anonymous) @ (index):4508
checkErrors @ (index):2126
loopCheckCompletion @ (index):2140
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
loopCheckCompletion @ (index):2142
setTimeout (async)
me.CreateShader @ (index):2144
EffectPass.NewShader @ (index):4493
Effect.NewShader @ (index):7593
(anonymous) @ (index):7750
Effect.Compile @ (index):7748
iLoadAndCompile @ (index):7974
iProcessShader @ (index):8030
(anonymous) @ (index):8115
setTimeout (async)
iInitShaders @ (index):8115
resultsInitStatic @ (index):8122
onload @ (index):8599
`

**Comment by TotoShampoin** [comment]
> Using firefox for shadertoy until this is fixed.

Same here. Also, I was starting to think it was a flatpak issue or something, but I'm glad I'm not the only one

**Comment by mvinaround** [comment]
Also still the same issue here.

**Comment by Treata11** [comment]
Seems like the browser is broken on all the platforms ... It's not just macOS.
Wonder what is taking Brave-devs so long to fix it

**Comment by btipling** [comment]
This still seems broken.

**Comment by kylegrover** [comment]
i'm about 3 days away from switching browsers because of this.

here's as far as i bothered to trace it before just using another browser:
![image](https://github.com/brave/brave-browser/assets/26558614/3d51d481-d362-44db-8437-01c2be42644d)

what should be happening:
![image](https://github.com/brave/brave-browser/assets/26558614/53a8ad19-70ab-42c3-8827-5b9a10c9155a)

you can see as well that a shader string is present, sourced, compiled, attached, and linked, and when you try `getAttachedShaders` at this point you get null 
![image](https://github.com/brave/brave-browser/assets/26558614/f4ee7032-6ce7-47f5-a366-e39d96f0a56d)

**Comment by wssbck** [comment]
I am facing the same issue. Commenting to increase visibility.

**Comment by LeonFedotov** [comment]
Since no brave people are brave enough to chip in perhaps we can discuss possible alternatives, hows firefox these days, is opera a good alternative?

**Comment by wssbck** [comment]
Not a dealbreaker for me, I will continue using Brave. To add more details:

[Version 1.65.126 Chromium: 124.0.6367.118 (Official Build) (64-bit)](https://brave.com/latest/)

**Comment by blamegames** [comment]
also having this issue.

**Comment by LeonFedotov** [comment]
@brave-dev 
![Screenshot 2024-05-06 at 20 46 48](https://github.com/brave/brave-browser/assets/130342/eb6a0227-ea3b-4cfa-9a47-54d4e06543e4)

**Comment by pierwmaria** [comment]
new to gls coding and shading in general but it's hard to start without some shadertoy renders...waiting for the debug too

**Comment by ImSimplySuperior** [comment]
still not working
bumping this

**Comment by njberman** [comment]
This is really bad, someone please fix this I have to use brave

**Comment by Hokiper** [comment]
> Since no brave people are brave enough to chip in perhaps we can discuss possible alternatives, hows firefox these days, is opera a good alternative?

Opera is the exact opposite of what probably induced you to use Brave. Firefox is fine, I think.

**Comment by LeonFedotov** [comment]
@BrendanEich duuuuudeeee!!!!!!!!!!!!

**Comment by mrmetaverse** [comment]
c'mon! 
![image](https://github.com/brave/brave-browser/assets/63426722/44b82fc0-e0f6-4eed-b7c7-19b71e137df0)

help us out 😅

**Comment by bdan-codes** [comment]
bump

**Comment by robiot** [comment]
![desert-tumbleweed](https://github.com/brave/brave-browser/assets/68228472/586a02c8-eb2e-4b2a-83e9-285ef1bd2d5a)

**Comment by LeonFedotov** [comment]
@rebron can you specify the timeline for fizing this please?

**Comment by arcin** [comment]
bump

**Comment by BaRaDa74** [comment]
there seems to be no interest as this hasnt even been triaged as of yet.  I've moved to Vivaldi, and there doesn't seem to be any issues so far.

**Comment by LeonFedotov** [comment]
More comments will get us to the front page of sort by comments maybe that
will help!

Leon. Everything web.
leonfedotov.com

**Comment by Laguna720** [comment]
Switched to Vivaldi because Brave doesnt want to acknowledge the issue

**Comment by johnpitchers** [comment]
Same. 
[Version 1.66.118 Chromium: 125.0.6422.147 (Official Build) (64-bit)]

**Comment by catboy-catfish** [comment]
Same thing here
[Version 1.66.118 Chromium: 125.0.6422.147 (Official Build) (64-bit)](https://brave.com/latest/)

> Switched to Vivaldi because Brave doesnt want to acknowledge the issue
Did that work for you?

**Comment by Laguna720** [comment]
> Same thing here [Version 1.66.118 Chromium: 125.0.6422.147 (Official Build) (64-bit)](https://brave.com/latest/)
> 
> > Switched to Vivaldi because Brave doesnt want to acknowledge the issue
> > Did that work for you?

Vivaldi works for me on Shadertoy. It also fixed a MPO issue I was having related to Windows.

**Comment by Miolith** [comment]
At least this issue is a nice way of listing some graphic programming enthusiasts

**Comment by TheJeran** [comment]
You know why I'm here

**Comment by fleity** [comment]
please fix the issues :-/

**Comment by ShinghoiXu** [comment]
plz fix this issue!!!

**Comment by jbunyadzade** [comment]
Please, fix this

**Comment by whoisnaisu** [comment]
follow up... I'm so sad it can't be use on brave

**Comment by robiot** [comment]
### Daily empty desert gif until they resolve this

Day 1

![335807698-586a02c8-eb2e-4b2a-83e9-285ef1bd2d5a](https://github.com/brave/brave-browser/assets/68228472/404aed96-b1f9-4de9-9888-0e3f553bbe04)

**Comment by brianpmaher** [comment]
Bump

**Comment by jakeembodiedlabs** [comment]
Same here. What's the story?

**Comment by Yona-Appletree** [comment]
pretty annoying now that I've been playing with shaders, what do we need to do to get attention?

**Comment by ShivanKaul** [comment]
For some reason, Brave always return `null` for `getAttachedShaders()` even if you attach one previously. This happens even without Shields on, so I don't think it's a fingerprinting protection issue. Still looking.

**Comment by jakeembodiedlabs** [comment]
> For some reason, Brave always return `null` for `getAttachedShaders()` even if you attach one previously. This happens even without Shields on, so I don't think it's a fingerprinting protection issue. Still looking.

I found the same thing. My current idea is making a Chrome plugin to fix it. If we find this can be fixed by modifying the JS, I can do that and post the source here (I have experience making Chrome plugins).

I also have tried to compile Brave in the past and failed to fully compile it into a build with a working installer, and found the entire process to be extremely complex. So, that is part of the reason behind my Chrome plugin idea. If anyone has experience building this from source and we can identify the fix then a PR would be appreciated.

**Comment by bsclifton** [comment]
@ShivanKaul has a fix up! Seems this was unintentionally caused by https://github.com/brave/brave-core/commit/19b9a04a06016dd1bbf0af824933a658bb13f20c#diff-d2b2184840d764ae72b93a75e08b8c7b87aea6013f622435fde429595805b9d9R25

**Comment by ShivanKaul** [comment]
Single-statement if blocks are the worst

**Comment by bsclifton** [comment]
Fix by @ShivanKaul merged; we'll be uplifting so the next release of Brave will have this patch 😄 Stay tuned

**Comment by Diffus3d** [comment]
Thank you very much for taking the time to solve this one.

**Comment by jakeembodiedlabs** [comment]
Thank you!!

**Comment by kjozwiak** [comment]
The above requires `1.67.129` or higher for `1.67.x` verification 👍

**Comment by zelfroster** [comment]
Thank god, Finally this is fixed 😌🚀

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.67.131 Chromium: 126.0.6478.126 (Official Build) (64-bit)
-- | --
Revision | e3ca6babc88232bab25830c09164a034adae83bd
OS | Windows 10 Version 22H2 (Build 19045.4529)
```
Using `1.67.123 Chromium: 126.0.6478.126` and the STR/Cases outlined via https://github.com/brave/brave-core/pull/24446#issue-2384800053 & https://github.com/brave/brave-browser/issues/37044#issue-2205323588, reproduced the original issue as per the following:

**`Example`** | **`Example`**
----------------|-----------------
![image](https://github.com/brave/brave-browser/assets/38657976/8107d0df-1d61-465e-a511-746802c05ca8) | ![image](https://github.com/brave/brave-browser/assets/38657976/1c345efa-874b-4968-be45-b5296d366e78)


Using the same STR/method as per above, verified that the issue has been resolved via `1.67.131 Chromium: 126.0.6478.126` as per the following:

**`Example`** | **`Example`**
----------------|-----------------
![image](https://github.com/brave/brave-browser/assets/38657976/9cecffd3-c7ea-423e-923a-053370001b3e) | ![image](https://github.com/brave/brave-browser/assets/38657976/1ffd20ab-16cc-4eb2-b0f4-f61743a2238e)

**Comment by catboy-catfish** [comment]
Tested this on windows 10 using the brave nightly build, works perfectly. Thanks guys
